openc3/lib/openc3/models/gem_model.rb

```diff
@@ -90,30 +80,33 @@ def self.install(name_or_path, scope:)
         Gem.done_installing_hooks.clear
         Gem.install(gem_file_path, "> 0.pre", :build_args => ['--no-document'], :prerelease => true)
       rescue => err
-        message = "Gem file #{gem_file_path} error installing to /gems\n#{err.formatted}"
+        message = "Gem file #{gem_file_path} error installing to #{ENV['GEM_HOME']}\n#{err.formatted}"
         Logger.error message
-      ensure
-        FileUtils.remove_entry(temp_dir) if temp_dir and File.exist?(temp_dir)
+        raise err
       end
     end
 
     def self.destroy(name)
-      bucket = Bucket.getClient()
-      Logger.info "Removing gem: #{name}"
-      bucket.delete_object(bucket: ENV['OPENC3_GEMS_BUCKET'], key: name)
       gem_name, version = self.extract_name_and_version(name)
-      begin
-        Gem::Uninstaller.new(gem_name, {:version => version, :force => true}).uninstall
-      rescue => err
-        message = "Gem file #{name} error uninstalling\n#{err.formatted}"
+      plugin_gem_names = PluginModel.gem_names
+      if plugin_gem_names.include?(name)
+        message = "Gem file #{name} can't be uninstalled because needed by installed plugin"
         Logger.error message
+        raise message
+      else
+        begin
+          Gem::Uninstaller.new(gem_name, {:version => version, :force => true}).uninstall
+        rescue => err
+          Logger.error "Gem file #{name} error uninstalling\n#{err.formatted}"
+          raise err
+        end
```

Comment:
Do we not delete it from the cache?

Reply:
The Gem::Uninstaller also removes from the cache.